tests/unit/clustering/kmeans_unittest.cc

```diff
@@ -11,9 +11,20 @@
 #include <shogun/clustering/KMeans.h>
 #include <shogun/clustering/KMeansMiniBatch.h>
 #include <shogun/distance/EuclideanDistance.h>
+#include <shogun/lib/observers/ParameterObserver.h>
+#include <shogun/lib/observers/ParameterObserverLogger.h>
 
 using namespace shogun;
 
+void check_consistency_observable(const CKMeans * kmeans, ParameterObserver * observer)
+{
+	auto total_observations = observer->get<int32_t>("num_observations");
+	auto observation = observer->get_observation(total_observations-1);
+	auto centers = observation->get<SGMatrix<float64_t>>("mus");
+
+	EXPECT_TRUE(centers.equals(kmeans->get<SGMatrix<float64_t>>("cluster_centers")));
```

Comment:
This only checks the last one?
Could you do a brief (internal, via print or so) check that the internal state matches the obs in every iteration and confirm?

This test is good though

Reply:
Yes, I'm checking the last one just to be sure that both the observer and model reach the same final values. Anyway, the internal state and the one observed by the observer match. We may need a better way to do these tests, since it would be really nice to have unit-tests that check the correctness of the observed values :)

Reply:
it is hard to do this in tests without modifying the algorithm code

BUT what we can do is this: you write a little MockMachine that has some very minimal state changes in the training loop, and then you observe those. Then we can assert the observations in a unit test. We should do this for primitive, SG*, and objects (where object observations are made of a single objects whose state is changed in the training loop in the mock machine)